### ISSUE TITLE
Release 4.3 introduces breaking changes in downstream apps when ran as async

### ISSUE THREAD

**Issue by tdj28** [deedy5/duckduckgo_search#180]
Moving from 4.2 to 4.3 results in this error in CrewAI, for example: https://github.com/joaomdmoura/crewAI/issues/193

RuntimeError: Cannot run the event loop while another loop is running

**Comment by tdj28** [comment]
First glance, looks like this line might be a potential source of the problem, added in version 4.3: https://github.com/deedy5/duckduckgo_search/blob/main/duckduckgo_search/duckduckgo_search.py#L13

**Comment by tdj28** [comment]
Quick update, it appears to only happen when the search call is being called in an async function, unable to replicate with a normal function.

Error does not occur here:

```
from langchain_community.tools import DuckDuckGoSearchRun

search = DuckDuckGoSearchRun()

print(search.run("Obama's first name?"))

```

does occur here:

```
from langchain_community.tools import DuckDuckGoSearchRun
import asyncio

async def main():
    search = DuckDuckGoSearchRun()

    print(search.run("Obama's first name?"))


asyncio.run(main())
```

**Comment by tdj28** [comment]
This is the proper way to call this async:

```
from langchain_community.tools import DuckDuckGoSearchRun
import asyncio

async def main():
    search = DuckDuckGoSearchRun()

    result = await asyncio.to_thread(search.run, "Obama's first name?")
    print(result)


asyncio.run(main())
```

Which is partially documented in the README. I'll close this issue as this does not look like a bug in the code, but it will be a breaking change (from 4.2 to 4.3) for users who prior used async without properly awaiting, but the error is theirs. Hopefully this issue will help anyone else experiencing this.

**Comment by deedy5** [comment]
The right way to use AsyncDDGS in asynchronous code.

**Comment by deedy5** [comment]
Fixed in v4.4
